SR2 Community Patch/scripts/server/objects/Civilian.as

```diff
@@ -434,6 +456,46 @@ tidy class CivilianScript {
 		return 0.2;
 	}
 
+	bool arriveWithEmpirePropulsionTechnology(Civilian& obj, vec3d enterDest) {
+		if(awaitingGateJump)
+			return obj.moveTo(enterDest, moveId, enterOrbit=false);
+
+		if(obj.owner.hasTrait(getTraitID("Jumpdrive"))) {
```

Comment:
This part has non-optional mechanical implications well beyond the scope of a simple bugfix. I don't think it's CP material.

Reply:
... To clarify, it's the jumpdrive/hyperdrive trait blocks.

Reply:
Yeah, thats a bit of a bummer. I didn't want to introduce any modifications either, but shouldn't that be base game?

When I fist started and had Hyperdrive, I wondered why my empire ships are so slow, then realized they don't FTL at all, they're just visual eye-candy made to look like it would belong there until u look closer...

I'm fine with removing the empire FTL, I wouldn't have added that if it wouldn't fix the annoying Heralds with their Beacon spam... 
The real problem (which not only affects Heralds) is: Pathfinding is used to determine reachability of trade areas, which is great imho. But it is mis-used by beacons and commerce stations (GlobalTradeNodes), where the pathfinder then thinks it's reachable directly and lets ships fly between all systems with beacons/CStations, as if they were connected.

Reply:
Regardless of if this is suitable for the CP, I'd like to borrow it and fit it into a research unlock for my own mod. I think one potential issue this has with the base game is if civilian ships can use jumpdrives to transport psionic reagents then that makes it virtually impossible to shoot down the trade ships and would make such a receiving planet nearly 100% immune to capture.

Reply:
Oh yeah, I'd like to add it to RS, too - and possibly extend it to colony ships. It just doesn't sound like a CP thing.

Reply:
Well, the jumpdrives on Civs are not as OP as u might think, they only jump intersystem paths. (try it, please ;)
Otherwise they wouldn't ever move normally ;) I have tested that, let's just say it's not good for game performance lol
They still move normally in systems/most of the time and can be shot down, I don't think they are protected by the effect like planets.

Consider this a minimal impact modification (MVP so to say), if u want to have that in RS/CE, I'd suggest a bit of a more involved diff... there are some things in the pathfinder I didn't want to touch here (to not add too many files), which would need changes too. I already made all this in IR, but thats too far away already to diff, with it's trade stuff.

Reply:
Btw, I would consider adding that to Colony ships a true mechanic changer, already thought about that. Those do not pathfind for a reason and there are small and large ones, they're slow for a reason, too. If u would give heralds jumpdrive colonists, you'd outrun mono :D

Reply:
I never said it was a *good* idea... 😛

Reply:
heh, yeah maybe there can be something done about them tho... the civs you really should try a game or two with. compare vanilla Heralds using their beacons to connect trade areas, with ships flying all around, with the jd version. or just plain Terrakin with their hyperdrive.

Reply:
Removed the FTL stuff and made a branch you can copy that from ;)
https://github.com/sol-oriens/SR2-Community-Patch/compare/Main...scitor:CivilianNavigationExt